### PR TITLE
Attempt to simplify/shorten GitHub CI commands

### PR DESCRIPTION
**Pull request by gregorbg**
The existing state of the CI was basically a remnant from our Travis migration.

Now that we have a much better idea of our own code organization and now that GitHub Actions has matured, I think we can clean this up a little bit.

### CODE REVIEW

**Comment by FinnIckler** [comment]
If you want to work on this without just waiting for the CI to see if it works, I recommend act https://github.com/nektos/act

**Comment by gregorbg** [comment]
Right now all the issues that are ocming up are issues that are failing fast (within ~1 minute) so I'm fine.
But thanks for the recommendation, looks interesting!

**Comment by gregorbg** [comment]
Also, depends on changes made in https://github.com/thewca/worldcubeassociation.org/pull/9674

**Review by FinnIckler — APPROVED** [review]
LGTM

File: app/models/result_validation_form.rb [hunk 1]
```diff
@@ -9,7 +9,6 @@ class ResultValidationForm
 
   COMP_VALIDATION_MODES = [["Pick competition(s) manually", COMP_VALIDATION_MANUAL], ["Execute for ALL competitions", COMP_VALIDATION_ALL]].freeze
 
-  ALL_COMPETITIONS_SCOPE = Competition.over.not_cancelled
```
Comment (FinnIckler) [inline app/models/result_validation_form.rb:12]: How did this even work. Wouldn't that variable only contain Competitions that were over/not_cancelled on boot?

Comment (gregorbg) [reply app/models/result_validation_form.rb:12]: No, actually Rails scopes are lazyily.
The thing that was failing was not the loading of actual data but the basic model validations like "which columns exist" etc. Rails wants to execute those as soon as you speak the name of any model, even if you're not fetching data.

**Pull request merged**